### DISCUSSION TITLE
スパコン上でのIGV使用について

### DISCUSSION THREAD

**Discussion by sc-fs** [nig-sc/nigsc_homepage2#285] [Q&A] [answered]
(sc-helpdeskへの質問を質問者の許可を得て代理で起票しております)

日頃より解析などでスパコンを使用しております。

遺伝研のスパコンですが、スパコン上でIGVは使用できますでしょうか？

いくつか、試してみたのですがうまくいかず、

もし可能でしたら実施方法をご教授いただけますでしょうか？

どうぞよろしくお願い申し上げます。

**Comment by sc-fs** [comment]
上手くいかない状況（エラー内容）が分からないので、
想定できるエラーの対処を記載します。

・実行するとjavaのエラーが発生する。
 この場合、以下を参照して対応してください。
 https://sc.ddbj.nig.ac.jp/software/java#%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A0%85

・「Xサーバ＊＊＊」のエラーが発生する
 igvはX11プロトコルでが画面表示をするようです。
 端末側でXサーバーが起動は済んでいますか？
 macはXQuartz、 windowsはXming など
 また、ゲートウェイノード接続時にX11プロトコルを許可設定にしてください。

・メモリが不足エラーが発生する。
 igvはメモリをデフォルトより多く必要とするようですので、qlogin時にメモリの指定
 をお願いします。

 実行例（展開したフォルダ上にあるigv.shを起動）
`$ qlogin -l s_vmem=12G -l mem_req=12G`
`$ export MALLOC_ARENA_MAX=3`
`$ IGV_Linux_2.16.0/igv.sh`

**Comment by nigyta** [comment, marked as answer]
以前のDDBJ講習会において、webブラウザ上で動作する igv-webapp をスパコンで動かしてポートフォワードを使って接続する方法を紹介していました。
https://www.youtube.com/watch?v=hXMVo1Yw-hA&list=PL_dbAF_dbOEorBr-so3ZFcqDMZWRCVVhB

また、同様にjbrowse2をスパコンで動かす方法についてもこちらで紹介されていました。
https://qiita.com/motthy/items/d8c295bbcfd8073fa6ab

ご参考までに
